sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1521,6 +1543,8 @@ class SessionCatalog(
     val qualifiedName = name.copy(database = database)
     functionRegistry.lookupFunction(name)
       .orElse(functionRegistry.lookupFunction(qualifiedName))
+      .orElse(tableFunctionRegistry.lookupFunction(name))
+      .orElse(tableFunctionRegistry.lookupFunction(qualifiedName))
```

Comment:
why do we look up the table function registry twice?

Reply:
Good point. Unlike non-table functions, there is currently no table function whose name can be qualified. We can remove the lookup function call with qualified name.